### PR TITLE
Trigger tip counter refresh on change

### PR DESCRIPTION
**Pull request by Tbaut**
closes https://github.com/polkadot-js/apps/issues/3730

Certainly not as nice as a subscription, but it does the job. Your call whether it's acceptable or not :)

tip-counter

### CODE REVIEW

File: packages/page-treasury/src/Tips/Tip.tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ interface Props {
   members: string[];
   onSelect: (hash: string, isSelected: boolean, value: BN) => void;
   onlyUntipped: boolean;
+  refresh: () => void;
```
Comment (jacogr) [inline packages/page-treasury/src/Tips/Tip.tsx:27]:
```suggestion
  onRefresh: () => void;
```

File: packages/page-treasury/src/Tips/Tip.tsx [hunk 2]
```diff
@@ -75,7 +76,7 @@ function extractTipState (tip: OpenTip | OpenTipTo225, hash: string, allAccounts
   };
 }
 
-function Tip ({ bestNumber, className = '', defaultId, hash, isMember, members, onSelect, onlyUntipped, tip }: Props): React.ReactElement<Props> | null {
+function Tip ({ bestNumber, className = '', defaultId, hash, isMember, members, onSelect, onlyUntipped, refresh, tip }: Props): React.ReactElement<Props> | null {
```
Comment (jacogr) [inline packages/page-treasury/src/Tips/Tip.tsx:79]:
```suggestion
function Tip ({ bestNumber, className = '', defaultId, hash, isMember, members, onRefresh, onSelect, onlyUntipped, tip }: Props): React.ReactElement<Props> | null {
```

File: packages/page-treasury/src/Tips/Tip.tsx [hunk 3]
```diff
@@ -150,6 +151,7 @@ function Tip ({ bestNumber, className = '', defaultId, hash, isMember, members,
               icon='times'
               isDisabled={!isFinder}
               label={t('Cancel')}
+              onSuccess={refresh}
```
Comment (jacogr) [inline packages/page-treasury/src/Tips/Tip.tsx:154]:
```suggestion
              onSuccess={onRefresh}
```

File: packages/page-treasury/src/Tips/Tip.tsx [hunk 4]
```diff
@@ -171,6 +173,7 @@ function Tip ({ bestNumber, className = '', defaultId, hash, isMember, members,
               accountId={councilId}
               icon='times'
               label={t<string>('Close')}
+              onSuccess={refresh}
```
Comment (jacogr) [inline packages/page-treasury/src/Tips/Tip.tsx:176]:
```suggestion
              onSuccess={onRefresh}
```

File: packages/page-treasury/src/Tips/Tips.tsx [hunk 5]
```diff
@@ -20,6 +20,7 @@ interface Props {
   isMember: boolean;
   members: string[];
   onSelectTip: (hash: string, isSelected: boolean, value: BN) => void,
+  refresh: () => void;
```
Comment (jacogr) [inline packages/page-treasury/src/Tips/Tips.tsx:23]:
```suggestion
  onRefresh: () => void;
```

File: packages/page-treasury/src/Tips/Tips.tsx [hunk 6]
```diff
@@ -43,7 +44,7 @@ function extractTips (optTips?: Option<OpenTip>[], hashes?: string[] | null): Ti
     );
 }
 
-function Tips ({ className = '', defaultId, hashes, isMember, members, onSelectTip }: Props): React.ReactElement<Props> {
+function Tips ({ className = '', defaultId, hashes, isMember, members, onSelectTip, refresh }: Props): React.ReactElement<Props> {
```
Comment (jacogr) [inline packages/page-treasury/src/Tips/Tips.tsx:47]:
```suggestion
function Tips ({ className = '', defaultId, hashes, isMember, members, onRefresh, onSelectTip }: Props): React.ReactElement<Props> {
```

File: packages/page-treasury/src/Tips/Tips.tsx [hunk 7]
```diff
@@ -91,6 +92,7 @@ function Tips ({ className = '', defaultId, hashes, isMember, members, onSelectT
           members={members}
           onSelect={onSelectTip}
           onlyUntipped={onlyUntipped}
+          refresh={refresh}
```
Comment (jacogr) [inline packages/page-treasury/src/Tips/Tips.tsx:95]:
```suggestion
          onRefresh={onRefresh}
```

File: packages/page-treasury/src/Tips/index.tsx [hunk 8]
```diff
@@ -78,6 +78,7 @@ function TipsEntry ({ className, hashes, isMember, members, trigger }: Props): R
         isMember={isMember}
         members={members}
         onSelectTip={_selectTip}
+        refresh={trigger}
```
Comment (jacogr) [inline packages/page-treasury/src/Tips/index.tsx:81]:
```suggestion
        onRefresh={trigger}
```

**Review by jacogr — COMMENTED** [review]
Yes, well, not sure there are any other options. The `entries()` just don't have subs. Unless there is a subscription to each, which is very wasteful.

TL;DR As ugly as it is, generally what I would have done as well.

Made some comments to `s/refresh/onRefresh/` (however that will probably drive the linter and alphabetical ordering crazy, so needs another adjust)

**Review by jacogr — APPROVED** [review]
Thank you
